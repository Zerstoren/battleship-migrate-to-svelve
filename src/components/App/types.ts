import { ILobbyStore } from '../../stores/lobby';

export enum GameStatus {
  MAIN = 'main',
  LOBBY_LIST = 'list',
  CREATE_LOBBY = 'create',
  WAIT_CONNECT = 'wait',
  SET_SHIPS = 'set_ships',
  GAME = 'game_process',
  GAMEOVER = 'game_loose',
  GAMEWIN = 'game_win',
}

export enum FireTurn {
  NOBODY = 'nobody',
  ME = 'me',
  OPPONENT = 'opponent',
}

export interface IGameState {
  gameStatus: GameStatus,
  gameWithUser: boolean,

  setGameStatus: (status: GameStatus) => void,
}

export interface IProps {
  handleChangeGameStatus: (startGame: GameStatus, lobby?: ILobbyStore | null) => void,
}

export interface ISetLobbyProps {
  handleSetLobby: (startGame: GameStatus, lobby: ILobbyStore) => void
}
