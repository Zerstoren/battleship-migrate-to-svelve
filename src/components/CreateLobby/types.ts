export interface FormErrors {
  lobbyName?: string,
  x?: string,
  y?: string,
  shipsCount?: string,
  ships4n?: string,
  ships3n?: string,
  ships2n?: string,
  ships1n?: string,
}

export interface FormInputData {
  lobbyName: string,
  x: string,
  y: string,
  ships4n: string,
  ships3n: string,
  ships2n: string,
  ships1n: string,
}
