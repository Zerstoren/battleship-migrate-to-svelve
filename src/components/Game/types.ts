import { IMainStore } from '../../stores/mainStore';

export interface IProps {
  mainStore?: IMainStore
}
