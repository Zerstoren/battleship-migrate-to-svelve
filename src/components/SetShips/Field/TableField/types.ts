import { DropTargetMonitor } from 'react-dnd';
import { DragObjectWithType } from 'react-dnd/lib/interfaces/hooksApi';

export enum MatrixFill {
  EMPTY = '',
  SHADOW = 's',
  ERR_SHADOW = 'e',
  SET = 'w',
  SET_KILL = 'k',
  MISS = 'm',
}

export type IMatrix = Array<Array<MatrixFill>>;

export interface DragObjectItem extends DragObjectWithType {
  size: number
}

export interface DropResult {
  item: DragObjectItem,
  monitor: DropTargetMonitor,
  result: boolean,
}
