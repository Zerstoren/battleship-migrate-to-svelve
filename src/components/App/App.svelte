<script lang="ts">
import { getContext } from 'svelte';
import type { IStore } from '../../stores';
import { STORE_CONTEXT } from '../../stores';
import CreateLobby from '../CreateLobby/CreateLobby.svelte';
import LobbyList from '../LobbyList/LobbyList.svelte';
import MainPage from '../MainPage/MainPage.svelte';
import SetShips from '../SetShips/SetShips.svelte';
import WaitConnect from '../WaitConnect/WaitConnect.svelte';
import { GameStatus } from './types';

const mainStore = getContext<IStore>(STORE_CONTEXT).mainStore;
let store = mainStore.store;

const handleAcceptError = () => {
  mainStore.setError('');
}

</script>

<div>
  {#if $store.error}
    <div class="alertError alert alert-warning alert-dismissible fade show">
      <strong>{$store.error}</strong>
      <input type="button" class="btn-close" onClick={handleAcceptError} />
    </div>
  {/if}
  <div class="centerDiv">
    {#if $store.status === GameStatus.MAIN}
      <MainPage />
    {:else if $store.status === GameStatus.CREATE_LOBBY}
      <CreateLobby />
    {:else if $store.status === GameStatus.WAIT_CONNECT}
      <WaitConnect />
    {:else if $store.status === GameStatus.LOBBY_LIST}
      <LobbyList />
    {:else if $store.status === GameStatus.SET_SHIPS}
      <SetShips />
    {/if}
  </div>
</div>

<style>
.centerDiv {
  width: 600px;
  margin: 0 auto;
  text-align: center;
}

.alertError {
  text-align: center;
}
</style>