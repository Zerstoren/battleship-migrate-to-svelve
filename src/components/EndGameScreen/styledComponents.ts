import styled from 'styled-components';

export const ButtonComplete = styled.button.attrs(() => ({
  className: 'btn btn-primary',
}))``;
