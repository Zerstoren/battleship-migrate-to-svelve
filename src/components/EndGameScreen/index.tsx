import React, { FC } from 'react';
import { inject } from 'mobx-react';
import { GameStatus } from '../App/types';
import { AppHeader } from '../../shared/StyledComponents/Headers';
import { IMainStore } from '../../stores/mainStore';
import { ButtonComplete } from './styledComponents';

interface IProps {
  mainStore?: IMainStore,
  gameResult: GameStatus,
}

const EndGameScreen: FC<IProps> = inject('mainStore')((props) => {
  const { gameResult, mainStore } = props;

  const handleToMainPage = () => {
    mainStore?.reset();
  };

  return (
    <>
      <AppHeader>{(gameResult === GameStatus.GAMEWIN ? 'You are win' : 'You are loose')}</AppHeader>
      <ButtonComplete onClick={handleToMainPage}>To main menu</ButtonComplete>
    </>
  );
});

export default EndGameScreen;
