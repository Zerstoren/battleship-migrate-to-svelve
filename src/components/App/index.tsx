import { inject, observer } from 'mobx-react';
import React, { FC } from 'react';
import { ILobbyStore } from '../../stores/lobby';
import { IMainStore } from '../../stores/mainStore';
import CreateLobby from '../CreateLobby';
import EndGameScreen from '../EndGameScreen';
import Game from '../Game';
import LobbyList from '../LobbyList';
import Main from '../MainPage';
import SetShips from '../SetShips';
import WaitConnect from '../WaitConnect';
import useWebsocketAppReconnect from './hooks';
import { AlertError, CenteredDiv } from './styledComponents';
import { GameStatus } from './types';

interface IAppProps {
  mainStore?: IMainStore
}

const App: FC<IAppProps> = inject('mainStore')(observer((props) => {
  const state = props.mainStore as IMainStore;

  useWebsocketAppReconnect(state);

  const handleGameStart = (startGame: GameStatus) => state?.setGameStatus(startGame);
  const handleSetLobby = (startGame: GameStatus, lobby: ILobbyStore) => state?.setLobby(startGame, lobby);
  const handleWaitLobby = (
    startGame: GameStatus,
    lobby: ILobbyStore | null = null,
  ) => state.setLobby(startGame, lobby);

  const handleAcceptError = () => state?.setError('');

  let component = null;

  switch (state.status) {
    case GameStatus.MAIN:
      component = (<Main handleChangeGameStatus={handleGameStart} />);
      break;

    case GameStatus.CREATE_LOBBY:
      component = (<CreateLobby handleChangeGameStatus={handleWaitLobby} />);
      break;

    case GameStatus.WAIT_CONNECT:
      component = (<WaitConnect />);
      break;

    case GameStatus.LOBBY_LIST:
      component = (<LobbyList handleSetLobby={handleSetLobby} />);
      break;

    case GameStatus.SET_SHIPS:
      component = (<SetShips />); break;

    case GameStatus.GAME:
      component = (<Game />);
      break;

    case GameStatus.GAMEWIN:
    case GameStatus.GAMEOVER:
      component = (<EndGameScreen gameResult={state.status as GameStatus} />);
      break;

    default:
      break;
  }

  let err = null;

  if (state?.error) {
    err = (
      <AlertError>
        <strong>{state?.error}</strong>
        <input type="button" className="btn-close" onClick={handleAcceptError} />
      </AlertError>
    );
  }

  return (
    <>
      {err}
      <CenteredDiv>
        {component}
      </CenteredDiv>
    </>
  );
}));

export default App;
