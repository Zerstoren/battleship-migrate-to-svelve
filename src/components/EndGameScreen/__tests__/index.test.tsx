import { mount } from 'enzyme';
import { Provider } from 'mobx-react';
import React from 'react';
import renderer from 'react-test-renderer';
import EndGameScreen from '..';
import { MainStore } from '../../../stores/mainStore';
import { GameStatus } from '../../App/types';

describe('EndGameScreen', () => {
  test('win screen screenshot', () => {
    const mainStore = MainStore.create({
      status: GameStatus.GAMEWIN
    });

    const test = renderer.create(<Provider mainStore={mainStore}><EndGameScreen gameResult={GameStatus.GAMEWIN} /></Provider>);
    expect(test.toJSON()).toMatchSnapshot();
  });

  test('loose screen screenshot', () => {
    const mainStore = MainStore.create({
      status: GameStatus.GAMEOVER
    });

    const test = renderer.create(<Provider mainStore={mainStore}><EndGameScreen gameResult={GameStatus.GAMEOVER} /></Provider>);
    expect(test.toJSON()).toMatchSnapshot();
  });

  test('reset store', () => {
    const mainStore = MainStore.create({
      status: GameStatus.GAMEWIN
    });

    const reset = jest.spyOn(mainStore, 'reset');
    const test = mount(<Provider mainStore={mainStore}><EndGameScreen gameResult={GameStatus.GAMEWIN} /></Provider>);
    test.find('button').simulate('click')
    expect(reset).toHaveBeenCalled();
    expect(mainStore.status).toBe(GameStatus.MAIN);
  });
});