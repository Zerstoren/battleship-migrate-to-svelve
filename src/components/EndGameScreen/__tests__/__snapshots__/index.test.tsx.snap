// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EndGameScreen loose screen screenshot 1`] = `
Array [
  <h1
    className="sc-bdfBwQ display-3"
  >
    You are loose
  </h1>,
  <button
    className="sc-gsTCUz btn btn-primary"
    onClick={[Function]}
  >
    To main menu
  </button>,
]
`;

exports[`EndGameScreen win screen screenshot 1`] = `
Array [
  <h1
    className="sc-bdfBwQ display-3"
  >
    You are win
  </h1>,
  <button
    className="sc-gsTCUz btn btn-primary"
    onClick={[Function]}
  >
    To main menu
  </button>,
]
`;
